app/src/main/java/org/openobservatory/ooniprobe/common/CountlyManager.java

```diff
@@ -0,0 +1,71 @@
+package org.openobservatory.ooniprobe.common;
+
+import android.content.Context;
+
+import com.google.android.gms.common.util.ArrayUtils;
+
+import org.openobservatory.engine.Engine;
+import org.openobservatory.ooniprobe.BuildConfig;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import ly.count.android.sdk.Countly;
+import ly.count.android.sdk.CountlyConfig;
+import ly.count.android.sdk.DeviceId;
+import ly.count.android.sdk.messaging.CountlyPush;
+
+public class CountlyManager {
+    private static String[] basicFeatures = new String[]{
+            ly.count.android.sdk.Countly.CountlyFeatureNames.location,
+    };
+
+    private static String[] analyticsFeatures = new String[]{
+            ly.count.android.sdk.Countly.CountlyFeatureNames.sessions,
+            ly.count.android.sdk.Countly.CountlyFeatureNames.views,
+            ly.count.android.sdk.Countly.CountlyFeatureNames.events
+            //TODO evaluate scrolls, clicks, forms, attribution
+    };
+
+    private static String[] crashFeatures = new String[]{
+            ly.count.android.sdk.Countly.CountlyFeatureNames.crashes,
+    };
+
+    private static String[] pushFeatures = new String[]{
+            ly.count.android.sdk.Countly.CountlyFeatureNames.push,
+    };
+
+    public static void register(Context ctx, PreferenceManager preferenceManager){
+        CountlyConfig config = new CountlyConfig()
+                .setAppKey("146836f41172f9e3287cab6f2cc347de3f5ddf3b")
+                .setContext(ctx)
+                .setDeviceId(Engine.getUUID())
+                .setRequiresConsent(true)
+                .setConsentEnabled(getConsentsEnabled(preferenceManager))
+                .setServerURL(BuildConfig.NOTIFICATION_SERVER)
+                .setLoggingEnabled(!BuildConfig.DEBUG)
+                .setViewTracking(true)
+                .setHttpPostForced(true)
+                .setViewTracking(true)
+                .enableCrashReporting();
+        Countly.sharedInstance().init(config);
+    }
+
+    public static String[] getConsentsEnabled(PreferenceManager preferenceManager) {
+        List<String> consents = new ArrayList(Arrays.asList(basicFeatures));
+        if (preferenceManager.isSendCrash())
+            consents.addAll(Arrays.asList(crashFeatures));
+        if (preferenceManager.isSendAnalytics())
+            consents.addAll(Arrays.asList(analyticsFeatures));
+        if (preferenceManager.isNotifications())
+            consents.addAll(Arrays.asList(pushFeatures));
+        return consents.toArray((new String[0]));
```

Comment:
https://stackoverflow.com/questions/4042434/converting-arrayliststring-to-string-in-java

The toArray() method without passing any argument returns Object[]. So you have to pass an array as an argument, which will be filled with the data from the list, and returned. You can pass an empty array as well, but you can also pass an array with the desired size.